src/Command/DrupalUpdate/UpdateScriptUtility.php

```diff
@@ -0,0 +1,317 @@
+          '.gitignore','.htaccess','CHANGELOG.txt','sites',
+      ];
+  }
+
+  /**
+   * Generate tabular view of available updates
+   * @param $version_detail
+   */
+  function printReleaseDetail($version_detail) {
+    $string_count = $this->getStringCountOfColumn($version_detail);
+    foreach ($version_detail as $key => $value){
+      echo str_repeat("-", array_sum($string_count)+6);
+      echo "\n";
+      echo "| ";
+      $i=0;
+      foreach ($value as $v){
+        if(is_array($v)){
+          foreach ($v as $t => $a){
+            echo $a . ",";
+          }
+          echo " | ";
+        } else{
+          $repetor = $string_count[$i]-strlen($v);
+          if($repetor > 0){
+            echo $v . str_repeat(" ", $string_count[$i]-strlen($v)) . " | ";
+          }else{
+            echo $v . str_repeat(" ", 1) . " | ";
+          }
+
+        }
+        $i++;
+      }
+      echo "\n";
+    }
+    echo "\n";
+  }
+
+  /**
+   * Update code based on available security update.
+   * @param $latest_security_updates
+   */
+
+  function updateCode($latest_security_updates) {
+    foreach ($latest_security_updates as $k => $value){
+      if(!isset($value['download_link'])){
+        continue;
+      }
+      if($value['package']=='drupal'){
+        $dirname = 'temp_drupal_core';
+        $filename = $value['file_path'] . "/" . $dirname . "";
+        if (!file_exists($filename)) {
+          $oldmask = umask(0);
+          mkdir($value['file_path'] . "/" . $dirname, 0777);
+          umask($oldmask);
+          $value['file_path'] = $value['file_path'] . "/" . $dirname . "";
+        } else {
+          echo "The directory $dirname exists.";
+        }
+      }
+      if(is_array($value['file_path'])){
+        foreach ($value['file_path'] as $item) {
+          $this->download_remote_file($value['package'], $value['download_link'], $item);
+        }
+      }else{
+        $this->download_remote_file($value['package'], $value['download_link'], $value['file_path']);
+      }
+
+    }
+
+  }
+
+  /**
+   * Download and extract tar files in given directory path.
+   * @param $package
+   * @param $file_url
+   * @param $save_to
+   */
+  function download_remote_file($package, $file_url, $save_to) {
+    if($package == 'drupal'){
+      $this->download_remote_file_drupal_core($package, $file_url, $save_to);
+      return;
+    }
+    $content = file_get_contents($file_url);
+    file_put_contents($save_to . '/' . $package . '.tar.gz', $content);
+    try {
+      $phar = new PharData($save_to . '/' . $package . '.tar.gz');
+      $this->rrmdir($save_to . '/' . $package);
+      $phar->extractTo($save_to, NULL, TRUE); // extract all files
+    } catch (\Exception $e) {
+      // handle errors
+    }
+  }
+
+  /**
+   * Download and extract tar files in drupal core temp directory path.
+   * @param $package
+   * @param $file_url
+   * @param $save_to
+   */
+  function download_remote_file_drupal_core($package, $file_url, $save_to) {
```

Comment:
How is this function different from download_remote_file? Is it possible to parameterise and combine them?